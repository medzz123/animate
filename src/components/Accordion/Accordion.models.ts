import { IconType } from 'react-icons';

export interface AccordionProps {
  title: string;
  id: string;
  Icon: IconType;
}
