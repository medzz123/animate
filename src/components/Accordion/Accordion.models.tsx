export interface AccordionProps {
  title: string;
  id: string;
}
