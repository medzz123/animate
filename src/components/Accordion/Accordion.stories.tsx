// import { Meta, Story } from '@storybook/react/types-6-0';
// import React from 'react';
// import { BiMove } from 'react-icons/bi';

// import Accordion from './Accordion';
// export default {
//   title: 'Components/Accordion',
//   component: Accordion,
// } as Meta;

// const Template: Story = (storyArguments) => (
//   <Accordion title="Accordion" id="accordion" Icon={BiMove} {...storyArguments}>
//     <p>Hello from the other side</p>
//   </Accordion>
// );

// export const AccordionStory = Template.bind({});
