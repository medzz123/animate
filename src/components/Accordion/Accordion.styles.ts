import KeyboardArrowDownIcon from '@material-ui/icons/KeyboardArrowDown';
import { Button, Header, Item, Panel, Root } from '@radix-ui/react-accordion';
import styled from 'styled-components';

export const StyledAccordion = styled(Root)``;

export const StyledItem = styled(Item)`
  border: 1px solid blue;
`;

export const StyledHeader = styled(Header)`
  margin: 0;
  display: 'flex';
`;

export const StyledButton = styled(Button)`
  background-color: transparent;
  border: none;
  padding: 10;
  flex: 1;
  display: flex;
  align-items: center;
  justify-content: space-between;
`;

export const StyledPanel = styled(Panel)`
  padding: 10;

  &[data-state='open'] {
    animation: slideDown 300ms ease-out;
  }

  &[data-state='closed'] {
    animation: slideUp 300ms ease-out;
  }

  @keyframes slideDown {
    from {
      height: 0;
    }

    to {
      height: var(--radix-accordion-panel-height);
    }
  }

  @keyframes slideUp {
    from {
      height: var(--radix-accordion-panel-height);
    }

    to {
      height: 0;
    }
  }
`;

export const AccordionChevron = styled(KeyboardArrowDownIcon)`
  transition: 'transform 300ms';
  [data-state='open'] & {
    transform: 'rotate(180deg)';
  }
`;
