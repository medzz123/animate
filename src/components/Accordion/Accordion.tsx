import { Header, Item, Root } from '@radix-ui/react-accordion';
import React, { FunctionComponent } from 'react';

import { AccordionProps } from './Accordion.models';
import {
  AccordionChevron,
  HorizontalLine,
  StyledButton,
  StyledPanel,
  Text,
} from './Accordion.styles';

const Accordion: FunctionComponent<AccordionProps> = (props) => {
  const { title, id, children, Icon } = props;

  return (
    <Root type="multiple">
      <Item key={id} value={id}>
        <Header>
          <StyledButton>
            <AccordionChevron aria-hidden />
            <Text>{title}</Text>
            <Icon />
            <HorizontalLine />
          </StyledButton>
        </Header>
        <StyledPanel>{children}</StyledPanel>
      </Item>
    </Root>
  );
};

export default Accordion;
