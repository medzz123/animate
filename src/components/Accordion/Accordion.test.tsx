import { render } from '@test/testUtils';
import React from 'react';

import Accordion from './Accordion';

test('Accordion Test', () => {
  const { container } = render(<Accordion id="title" title="test" />);

  expect(container).toBeInTheDocument();
});
