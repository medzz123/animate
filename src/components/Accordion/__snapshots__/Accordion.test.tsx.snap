// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Accordion Test 1`] = `
<div>
  <div
    class="sc-bdfBwQ"
    type="multiple"
  />
</div>
`;
